service/selectors/entities/users.js

```diff
@@ -19,6 +33,46 @@ export const getCurrentUser = createSelector(
     }
 );
 
-export function getUser(state, id) {
-    return state.entities.users.profiles[id];
-}
+export const getProfileSetInCurrentChannel = createSelector(
+    getCurrentChannelId,
+    getProfilesInChannel,
+    (currentChannel, channelProfiles) => {
+        return channelProfiles[currentChannel];
+    }
+);
+
+export const getProfilesInCurrentChannel = createSelector(
+    getUsers,
+    getUserStatuses,
+    getProfileSetInCurrentChannel,
+    (profiles, statuses, currentChannelProfileSet) => {
+        const currentProfiles = [];
+        if (typeof currentChannelProfileSet === 'undefined') {
+            return currentProfiles;
+        }
+
+        currentChannelProfileSet.forEach((p) => {
+            currentProfiles.push({
+                ...profiles[p],
+                status: statuses[p]
+            });
+        });
+
+        // We could get rid of this if server side sorting is a possibility
+        const sortedCurrentProfiles = currentProfiles.sort((a, b) => {
+            const nameA = a.first ? a.first : a.username;
```

Comment:
We'll need the display name here as well. Also, you have the wrong field name here (`first` should be `first_name`)